File: db4on/tutorial/vb/db4o-tutorial-chapters/src/com/db4o/f1/chapter4/TemperatureSensorReadout.vb
Imports System
Namespace com.db4o.f1.chapter4
	Public Class TemperatureSensorReadout
	Inherits SensorReadout
		Private _temperature As Double

		Public Sub New(ByVal time As DateTime, ByVal car As Car, ByVal description As String, ByVal temperature As Double)
			MyBase.New(time, car, description)
			_temperature = temperature
		End Sub 

		Public ReadOnly Property Temperature() As Double
			Get
				Return _temperature
			End Get
		End Property 

		Public Overloads Overrides Function ToString() As String
			Return String.Format("{0} temp: {1}", MyBase.ToString(), _temperature)
		End Function

	End Class
End Namespace

File: tutorial/src/com/db4o/f1/chapter7/TemperatureSensorReadout.java
package com.db4o.f1.chapter7;

import java.util.*;


public class TemperatureSensorReadout extends SensorReadout {
    private double temperature;
    
    public TemperatureSensorReadout(
            Date time,Car car,
            String description,double temperature) {
        super(time,car,description);
        this.temperature=temperature;
    }
    
    public double getTemperature() {
        activate();
        return temperature;
    }

    public String toString() {
        return super.toString()+" temp : "+temperature;
    }
}
